modules/ROOT/pages/salesforce/salesforce-connector-use-cases.adoc

```diff
@@ -0,0 +1,359 @@
+}]]]></ee:set-payload>
+      </ee:message>
+    </ee:transform>
+    <salesforce:upsert config-ref="Salesforce_Sfdc_config"
+    externalIdFieldName="Id" type="Account" doc:name="Upsert" />
+    <ee:transform doc:name="Transform Message" >
+      <ee:message >
+        <ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+payload map {
+  id:$.id,
+  errors:$.errors,
+  success:$.success,
+  created:$.created
+
+  }]]></ee:set-payload>
+      </ee:message>
+    </ee:transform>
+  </flow>
+  <flow name="find_duplicates_for_account_flow" >
+    <http:listener config-ref="HTTP_Listener_config" path="/findDuplicates" doc:name="Listener" />
+    <ee:transform doc:name="Transform Message" >
+      <ee:message >
+        <ee:set-payload ><![CDATA[%dw 2.0
+output application/java
+---
+[
+  payload
+]]]></ee:set-payload>
+      </ee:message>
+    </ee:transform>
+    <salesforce:find-duplicates config-ref="Salesforce_Sfdc_config" type="Account"
+    doc:name="Find duplicates" />
+    <ee:transform doc:name="Transform Message" >
+      <ee:message >
+        <ee:set-payload ><![CDATA[%dw 2.0
+output application/json
+---
+{
+  success: payload.success,
+  duplicateResults: {
+    (payload.duplicateResults map {
+      matchRecords: $.matchResults
+    }
+    )
+  },
+  duplicateRuleEntityType: payload.duplicateRuleEntityType,
+  duplicateRule: payload.duplicateRule,
+  allowSave: payload.allowSave,
+  errorMessage: payload.errorMessage
+}]]></ee:set-payload>
+      </ee:message>
+    </ee:transform>
+  </flow>
+  <flow name="crud-appFlow" >
+    <http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/"/>
+    <salesforce:convert-lead doc:name="Convert lead" config-ref="Salesforce_Sfdc_config"/>
+  </flow>
+</mule>
+----
+
+[[parentchild]]
+== Example: Create or Update an Object With Parent Child Relationships
+
+A Salesforce object can have standard or custom relationships between objects.
+
+The relationships between the objects are usually one-to-many parent child relationships,
+but can be any link between two objects residing in Salesforce.
+
+Creating or altering objects with relationships is challenging. This section shows how to perform an upsert for an object using the Salesforce connector.
+
+Create a structure in Salesforce for this relationship.
+
+This example assumes two custom types: `MyCustomObject` and `MyOtherCustomObject`.
+
+MyCustomObject must hold a relationship to a MyOtherCustomObject. When upserting a `MyCustomObject`, the POJO sent as input to the Salesforce connector looks like this:
+
+[source,json,linenums]
+----
+{
+  ...
+  // MyCustomObject's fields ...
+  OtherObject__r:
+  {
+    CustomField__c : 'ABC123',
+    type: 'MyOtherCustomObject__c'
+  }
+}
+----
+
+`OtherObject` is the name of `MyCustomObject` field whose value must be a reference to a `MyOtherCustomObject` object. `OtherObject__r` indicates the name of the field that is set and that its a relationship to another object.
```

Comment:
This is hard to understand. (also you need a "the" before `MyCustomObject` field.) xxx__r is the API Name of the relationship field, and xxx__c is the API Name of the custom field. Are we mixing up the names of things with the things, or...? It would also help if the examples did *not* use the word custom object, it just makes everything harder to talk about.